library/data/registry_entry_attachment/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentImpl.java

```diff
@@ -176,6 +176,10 @@ public Iterator<TagEntry<R, V>> tagEntryIterator() {
 
 	@Override
 	public void put(R entry, V value) {
+		if (this.registry.getId(entry) == null) {
```

Comment:
I just realized something: this check does not work with defaulted registries, such as entity types.
Is there a `Registry.contains` method?